### PR TITLE
BLD: Bump the build number to pick up latest arrow

### LINKED ISSUE #42: Switch to dynamic linking for boost-cpp
see also https://github.com/conda-forge/arrow-cpp-feedstock/issues/47

### PR DESCRIPTION
**Pull request by cpcloud**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)

Closes #42

### CODE REVIEW

**Comment by cpcloud** [comment]
We are still getting warnings about not finding libc, libgcc_s, libstd++, boost, arrow, and libm:

```
WARNING (parquet-cpp,lib/libparquet.so.1.4.0): did not find - or even know where to look for: /lib64/libc.so.6
WARNING (parquet-cpp,lib/libparquet.so.1.4.0): Needed DSO lib/libboost_regex.so.1.67.0 found in ['boost-cpp']
WARNING (parquet-cpp,lib/libparquet.so.1.4.0): .. but ['boost-cpp'] not in reqs/run, i.e. it is overlinked (likely) or a missing dependency (less likely)
WARNING (parquet-cpp,lib/libparquet.so.1.4.0): lib/libgcc_s.so.1 not found in any packages
WARNING (parquet-cpp,lib/libparquet.so.1.4.0): lib/libstdc++.so.6 not found in any packages
WARNING (parquet-cpp,lib/libparquet.so.1.4.0): lib/libarrow.so.0 not found in any packages
WARNING (parquet-cpp,lib/libparquet.so.1.4.0): did not find - or even know where to look for: /lib64/libm.so.6
```

**Comment by cpcloud** [comment]
@xhochy Shouldn't `boost-cpp` be a run dependency (in addition to a build dependency) because we're linking dynamically to boost here?

**Comment by xhochy** [comment]
@cpcloud yes, it should be. We forgot that when we switched to dynamic linking.

**Comment by cpcloud** [comment]
Ok, I'll add.

**Comment by cpcloud** [comment]
@xhochy I'm not sure why conda is complaining that it can't find the SOs, even though they are clearly available.

Moreover, when I build this locally I only get warnings about libc and libm (which makes sense since those are not controlled by conda).

**Comment by cpcloud** [comment]
Another thing, do we actually need a C compiler in the build deps? Is parquet-cpp transitively compiling any C code?

**Comment by cpcloud** [comment]
It appears that if we don't include `{{ compiler('c') }}` in the build deps we get missing SO warnings about libgcc_s and libgcc-ng.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -18,13 +18,14 @@ source:
...
 build:
-  number: 1
+  number: 2
   skip: true  # [win32]
   skip: true  # [win and py<35]
 
 requirements:
   build:
     - cmake
+    - patchelf
```
Comment (xhochy) [inline recipe/meta.yaml]: @cpcloud can you remove this to get a green build again?

Comment (cpcloud) [reply recipe/meta.yaml]: yep

**Comment by cpcloud** [comment]
@xhochy Thanks for merging!

**Pull request merged**